### ISSUE TITLE
Clarify instructions in readme

### ISSUE THREAD

**Issue by brittag** [yyuu/pyenv#481]
Just a few documentation comments as I work through installing and trying to use this:

**Homebrew on Mac OS X**

In the readme's [Homebrew on Mac OS X](https://github.com/yyuu/pyenv#homebrew-on-mac-os-x) instructions, it would help for "After installation, you'll still need to add eval "$(pyenv init -)" to your profile (as stated in the caveats)." to be more specific about where "the caveats" are. Reading this as a newcomer, I felt a little confused at that point, since I hadn't seen a section titled "caveats".

Does this really mean "now go to [Basic GitHub Checkout](https://github.com/yyuu/pyenv#basic-github-checkout) and follow those instructions starting at step 3"? If you install pyenv via Homebrew, what is the "$PYENV_ROOT" that you should use for step 5?

**Basic GitHub Checkout**

The readme could also explain in step 5 where and how to "download and unpack the source" if that's a necessary step (when I tried this, I typed pyenv install 3.4.3, and it seemed to download and install that version without requiring a separate download/unpack step in preparation).

At the end of this installation section, it would be nice to point to a section that provides the most likely next steps for the reader.

**Choosing the Python Version**

The [Choosing the Python Version](https://github.com/yyuu/pyenv#choosing-the-python-version) section links to #pyenv-shell, #pyenv-local, and #pyenv-global, but those links don't go anywhere right now. It looks like they should link to sections of https://github.com/yyuu/pyenv/blob/master/COMMANDS.md instead.

**Comment by yyuu** [comment]
This is the caveats of brew's pyenv formula. I'm happy if you gave me back some PR to improve the documentation.

https://github.com/Homebrew/homebrew/blob/34776235339f59a01e996b29de6de76dd64de2eb/Library/Formula/pyenv.rb#L34-L41

**Comment by brittag** [comment]
Thanks, I must have missed that caveats section if it displayed during the Homebrew-based installation!

```To enable shims and autocompletion add to your profile:
  if which pyenv > /dev/null; then eval "$(pyenv init -)"; fi

To use Homebrew's directories rather than ~/.pyenv add to your profile:
  export PYENV_ROOT=/usr/local/var/pyenv```

Looking at this, it would be great to improve this to the level of detail that the "Basic GitHub Checkout" section has. A newcomer might not know what "your profile" means, so the copy-and-pastable `echo 'eval "$(pyenv init -)"' >> ~/.bash_profile` style commands are really helpful. You may also want to include a link to https://github.com/yyuu/pyenv to explain "shims and autocompletion". :)

In order to submit a pull request improving all these things, I would need to understand this project a little better. I have a couple questions to help me suggest the correct information:

1. If a person does the Homebrew install process, what are the next steps for them after following the caveats?

2. Am I correct that a separate  "download and unpack the source" step is not necessary?

**Comment by yyuu** [comment]
1. If a brew user setup pyenv in one's "shell profile" as mentioned in the caveats, s?he will need to restart his/her shell session to make it applied.
2. Good catch. You're correct. The sentence of "download and unpack the source" is just unnecessary. That's what pyenv will do automatically....

**Comment by yyuu** [comment]
Solved by #485
